arm-servicefabric/2016-09-01/swagger/servicefabric.json

```diff
@@ -0,0 +1,757 @@
+          ]
+        },
+        "upgradeMode": {
+          "type": "string",
+          "enum": [
+            "Default",
+            "Automatic",
+            "Manual"
+          ]
+        },
+        "clientCertificateThumbprints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientCertificateThumbprint"
+          }
+        },
+        "clientCertificateCommonNames": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientCertificateCommonName"
+          }
+        },
+        "fabricSettings": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SettingsSectionDescription"
+          }
+        },
+        "reverseProxyCertificate": {
+          "$ref": "#/definitions/CertificateDescription"
+        },
+        "managementEndpoint": {
+          "type": "string",
+          "readOnly": true
+        },
+        "nodeTypes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NodeTypeDescription"
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Default",
+            "WaitingForNodes",
+            "Deploying",
+            "BaselineUpgrade",
+            "UpdatingUserConfiguration",
+            "UpdatingUserCertificate",
+            "UpdatingInfrastructure",
+            "EnforcingClusterVersion",
+            "UpgradeServiceUnreachable",
+            "Deleting",
+            "ScaleUp",
+            "ScaleDown",
+            "AutoScale",
+            "Ready",
+            "Failed"
+          ]
+        },
+        "vmImage": {
+          "type": "string",
+          "readOnly": true
+        },
+        "diagnosticsStorageAccountConfig": {
+          "$ref": "#/definitions/diagnosticsStorageAccountConfig"
+        },
+        "upgradeDescription": {
+          "$ref": "#/definitions/paasClusterUpgradePolicy"
+        }
+      }
+    },
+    "clusterPatchParameters": {
+      "description": "Cluster patch parameters",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/clusterResourceProperties"
+        },
+        "propertyNames": {
+          "description": "Allowed properties",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": [
+              "ReliabilityLevel",
+              "ClusterUpgradeMode",
+              "ClientCertificateThumbprints",
+              "FabricSettings",
+              "ReverseProxyCertificate",
+              "NodeTypes"
+            ]
+          }
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": false
+    },
+    "ClusterPatchRequest": {
+      "description": "Cluster patch request",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/clusterPatchParameters"
+        }
+      }
+    },
```

Comment:
Do you think we can provide a better name for the type `ClusterPatchRequest` ?  with this name the prototype of the update method will looks like

```java
class Clusters {
  Cluster Update(String resourceGroupName, String customerName, ClusterPatchRequest clusterPatchRequest);
}

```

How about re-naming `ClusterPatchRequest` to `ClusterUpdateParameters` ? and the nested `ClusterPatchParameters` can be named as `ClusterPropertiesUpdateParameters`. This name seems better match with the operation name (Clusters.Update).

In addition to this, we could apply `x-ms-client-flatten` extension https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-client-flatten here as well for simple user experience (Ref: https://github.com/Azure/azure-rest-api-specs/pull/984/files#r104088396)